### ISSUE TITLE
All-in-One WPS

### ISSUE THREAD

**Issue by fenris** [flathub/com.wps.Office#65]
I've installed the WPS then I tried to change it to make it available to open all-in-one. Then after restarting the WPS ... it doesn't run anymore. 

How to revert back to WPS modules instead of all-in-one?

**Comment by gasinvein** [comment]
Try removing lines
```
wpsoffice\Application%20Settings\AppComponentMode=prome_fushion
wpsoffice\Application%20Settings\AppComponentModeInstall=prome_fushion
```
from `~/.var/app/com.wps.Office/config/Kingsoft/Office.conf` file.

**Comment by fwolf** [comment]
By the way, removing this 2 line of config, also solved my wps missing titlebar problem, and newest version won't start problem !

In Ubuntu, it in` ~/.config/Kingsoft/Office.conf` .

**Comment by dwuku** [comment]
Is it possible to actually make AIO work?

**Comment by gasinvein** [comment]
@dwuku Does it work in non-flatpak package?

**Comment by dwuku** [comment]
Yes, works fine. I was using 11.1.0.9080 deb package before switching to the flatpak

**Comment by gasinvein** [comment]
Looks like WPS Office all-in-one mode doesn't work when `XDG_CONFIG_HOME` is set, even if it point to the default path (`$HOME/.config`). Not sure if this can be easily fixed on our side.

**Comment by dwuku** [comment]
@gasinvein just out of curiosity, how did you figure this out?

**Comment by gasinvein** [comment]
@dwuku I've tried to reproduce flatpak behavior in non-flatpak environment, e.g. set `XDG_*_HOME` vars, which are usually unset on host.

**Comment by gasinvein** [comment]
We can workaround this by symlinking `~/.cache` to `$XDG_CONFIG_HOME` and then unsetting `XDG_CONFIG_HOME` upon startup, but this can break things for users who've added filesystem overrides, especially `--filesystem=home`.

**Comment by dwuku** [comment]
Symlinking ~/.config did not work for me. But interestingly, if you do not touch anything else and only unset XDG_CONFIG_HOME by entering a shell inside the flatpak, AIO mode does start working. Of course as soon as you exit the shell the config files are deleted (I think the new ones are created in /Kingsoft ?)

**Comment by gasinvein** [comment]
@dwuku  I've meant symlinking `~/.config` _and_ unsetting `XDG_CONFIG_HOME` both inside flatpak sandbox, surely.

**Comment by dwuku** [comment]
@gasinvein could you please elaborate a bit on how to create this symlink? I create a filesystem override to ~/.config first?

**Comment by gasinvein** [comment]
@dwuku I'll open a PR with modified wrapper once/if I figure out how to do this safely.
>I create a filesystem override to ~/.config first?

No! You absolutely shouldn't create filesystem overrides like this.

**Comment by dwuku** [comment]
@gasinvein alright thanks, but could you still answer the question please? Just for knowledge. What did you mean by symlinking `~/.config` to `$XDG_CONFIG_HOME` ?

**Comment by gasinvein** [comment]
@dwuku By default, in the flatpak sandbox `/home` is on a per-instance tmpfs, so normally `~/.config` doesn't exist in the sandbox. Insted, a special directory `~/.var/app/com.wps.Office/config` is mounted from host, and `XDG_CONFIG_HOME` env var points to it.
My idea (not sure if it's a good one, though) was to add to the `wps.sh` wrapper something like  
`ln -s $XDG_CONFIG_HOME $HOME/.config` followed by `unset XDG_CONFIG_HOME`,  
so that we'll end up with config files still stored in `~/.var/app/com.wps.Office/config`, but without `XDG_CONFIG_HOME` env var set.
Doing so is unsafe, since user may have filesystem overrides, so we'll have to check if `~/.config` doesn't exist in sandbox, first.

**Comment by dwuku** [comment]
@gasinvein ah got it, inside the sandbox! For some reason I was thinking about the ~/.config of the host. Thanks so much for explaining in detail. 

This workaround indeed does the job, I gave it a try just now. 

The behaviour is a bit confusing though.
Once that symlink is created, the application does work fine, but the wrapper throws a message in the beginning which says "No config exists, creating one" and when you do `ls /` there is a new Kingsoft directory (which is otherwise not created when `XDG_CONFIG_HOME` is not unset)

**Comment by gasinvein** [comment]
@dwuku The wrapper needs to be modified to handle this correctly.

**Comment by dwuku** [comment]
@gasinvein oh right, yes.

Also, I'm wondering if [this](https://github.com/flatpak/flatpak/issues/867#issuecomment-311595653) could be useful in any way. Although, goes against sandboxing principles I guess.

**Comment by gasinvein** [comment]
I think it was fixed by latest upstream update, I can't reproduce it since of 11.1.0.10161
Can you please re-test if the issue is still present?

**Comment by okkymabruri** [comment]
Hi Guys, I got this issue on my machine, ubuntu 21.04 with WPS 11.1.0.10161
Could you tell me how to solve this issue?
@gasinvein

**Comment by gasinvein** [comment]
@okkymabruri I can't reproduce it anymore, but you still can try [this](https://github.com/flathub/com.wps.Office/issues/65#issuecomment-596127095).

**Comment by okkymabruri** [comment]
@gasinvein , I have tried it on
~/.config/Kingsoft/Office.conf
~/.var/app/com.wps.Office/config/Kingsoft/Office.conf
but still doesn't work

**Comment by Sciroccogti** [comment]
> In Ubuntu, it in` ~/.config/Kingsoft/Office.conf` .

In my Ubuntu 22.10, the file does not exist. Is it placed somewhere else?

**Comment by fwolf** [comment]
I'm using debian sid now, and the conf file is still there, dont know Ubuntu 22.10 change it or not.

> @fwolf
> 
> > In Ubuntu, it in` ~/.config/Kingsoft/Office.conf` .
> 
> In my Ubuntu 22.10, the file does not exist. Is it placed somewhere else?

**Comment by Sciroccogti** [comment]
OK I find it in `~/.var/app/com.wps.Office/config/Kingsoft/Office.conf`.

In my case, the two lines mentioned by @gasinvein do not exist originally, and manually adding them will enable all-in-one mode. That's great.

> I'm using debian sid now, and the conf file is still there, dont know Ubuntu 22.10 change it or not.
> 
> > @fwolf
> > > In Ubuntu, it in` ~/.config/Kingsoft/Office.conf` .
> > 
> > 
> > In my Ubuntu 22.10, the file does not exist. Is it placed somewhere else?
